catalog/core/catalog-core-solr/src/test/java/ddf/catalog/source/solr/SolrProviderRealTimeQueryTest.java

```diff
@@ -261,4 +280,42 @@ public void testRealTimeQueryNotId() throws Exception {
     // Verify a "not ID query" does not return the result.
     queryAndVerifyCount(0, filter, provider);
   }
+
+  @Test
+  public void testCommitNrt() throws Exception {
+
+    final String metacardTitle = "testCommitNrt";
+
+    deleteAll(provider);
+
+    MetacardType nrtMetacardType =
+        new MetacardTypeImpl(
+            COMMIT_NRT_TYPE, MetacardImpl.BASIC_METACARD.getAttributeDescriptors());
+
+    MockMetacard metacard = new MockMetacard(Library.getFlagstaffRecord(), nrtMetacardType);
+    metacard.setTitle(metacardTitle);
+
+    CreateResponse createResponse = create(metacard, provider);
+    Optional<String> id =
+        createResponse.getCreatedMetacards().stream().map(Metacard::getId).findFirst();
+
+    if (!id.isPresent()) {
```

Comment:
✏️ Using `com.github.npathai.hamcrestopt.OptionalMatchers`, this could be written as:
```suggestion
    assertThat(id, OptionalMatchers.isPresent());
```